aexpect/client.py

```diff
@@ -852,10 +874,12 @@ def read_until_output_matches(self, patterns, filter_func=lambda x: x,
             if not poll_status:
                 raise ExpectTimeoutError(patterns, output)
             # Read data from child
-            data = self.read_nonblocking(internal_timeout,
-                                         end_time - time.time())
-            if not data:
+            read, data = self._read_nonblocking(internal_timeout,
+                                                end_time - time.time())
+            if not read:
```

Comment:
it sounds like this line need put behind line 882 if this PR wants to satisfy requirement in https://github.com/avocado-framework/aexpect/pull/78.

Reply:
It should be the opposite way around, `read` is the length of the raw bytes read while the `data` is the decoded (with ignore) string.

Therefore when the pipe is closed, both `read` and `data` will be false (`0` and `''`). When the read is solely composed of ignore chars, the `read` will be non-zero (amount of read bytes, at least `1`) but `data` will be empty (`''`).

Reply:
`read >= len(data)`

Reply:
thank for explanation